modules/core/src/test/java/org/apache/ignite/internal/processors/cache/CacheMetricsTouchTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache;
+
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.processors.cache.store.CacheStoreManager;
+import org.apache.ignite.internal.processors.metric.GridMetricManager;
+import org.apache.ignite.internal.processors.metric.MetricRegistry;
+import org.apache.ignite.internal.processors.metric.impl.AtomicLongMetric;
+import org.apache.ignite.internal.processors.metric.impl.MetricUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Before;
+import org.junit.Test;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Test class for cache touch metrics.
+ */
+
+public class CacheMetricsTouchTest extends GridCommonAbstractTest {
```

Comment:
This test is not included in any test suite.
Perhaps, it would be nice to add integration tests as well. Please take a look at `GridCacheAbstractMetricsSelfTest`